### PR TITLE
Anticipate div by zero in FocalLoss normalization

### PR DESCRIPTION
**Pull request by CharlesGaydon**

### CODE REVIEW

File: semantic_val/models/model.py [hunk 1]
```diff
@@ -49,7 +50,10 @@ def forward(self, logits, targets):
             pi = proba[:, i] * ti
             ai = self.alpha[i]
             loss += -(ti * ai) * (1 - pi) ** self.gamma * torch.log(pi + self.eps)
-        loss = loss.sum() / (targets == self.rare_class_dim).sum()
+        n_points = logits.size(0)
+        n_points_rare_class = (targets == self.rare_class_dim).sum()
+        normalization_factor = (n_points_rare_class + n_points / 100) / 2
+        loss = loss.sum() / normalization_factor
```
Comment (portematthieu) [inline semantic_val/models/model.py:56]: You're giving less importance to samples with many points of the rare class ?

Comment (portematthieu) [reply semantic_val/models/model.py:56]: I think the normalization factor should be the inverse of the current one.
